src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/HttpApi/ApiClient.cs

```diff
@@ -124,6 +131,112 @@ private async Task<Models.ProcessInfo> GetProcessAsync(string processKey, Cancel
+        }
+
+        private async Task<Dictionary<string, string>> GetProcessEnvironmentAsync(string processKey, CancellationToken token)
+        {
+            using HttpRequestMessage request = new(HttpMethod.Get, $"/processes/{processKey}/env");
+            request.Headers.Add(HeaderNames.Accept, ContentTypes.ApplicationJson);
+
+            using HttpResponseMessage response = await SendAndLogAsync(
+                request,
+                HttpCompletionOption.ResponseContentRead,
+                token).ConfigureAwait(false);
+
+            switch (response.StatusCode)
+            {
+                case HttpStatusCode.OK:
+                    ValidateContentType(response, ContentTypes.ApplicationJson);
+                    return await ReadContentAsync<Dictionary<string, string>>(response).ConfigureAwait(false);
+                case HttpStatusCode.BadRequest:
+                    ValidateContentType(response, ContentTypes.ApplicationProblemJson);
+                    throw await CreateValidationProblemDetailsExceptionAsync(response).ConfigureAwait(false);
+                case HttpStatusCode.Unauthorized:
+                case HttpStatusCode.NotFound:
+                    ThrowIfNotSuccess(response);
+                    break;
+            }
+
+            throw await CreateUnexpectedStatusCodeExceptionAsync(response).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Get /dump/{pid}?type={dumpType}
+        /// </summary>
+        public Task<ApiStreamHolder> CaptureDumpAsync(int pid, DumpType dumpType, CancellationToken token)
+        {
+            return CaptureDumpAsync(pid.ToString(CultureInfo.InvariantCulture), dumpType, token);
+        }
+
+        /// <summary>
+        /// Get /dump/{uid}?type={dumpType}
+        /// </summary>
+        public Task<ApiStreamHolder> CaptureDumpAsync(Guid uid, DumpType dumpType, CancellationToken token)
+        {
+            return CaptureDumpAsync(uid.ToString("D"), dumpType, token);
+        }
+
+        private async Task<ApiStreamHolder> CaptureDumpAsync(string processKey, DumpType dumpType, CancellationToken token)
+        {
+            using HttpRequestMessage request = new(HttpMethod.Get, $"/dump/{processKey}?type={dumpType.ToString("G")}");
+            request.Headers.Add(HeaderNames.Accept, ContentTypes.ApplicationOctetStream);
+
+            using DisposableBox<HttpResponseMessage> responseBox = new(
+                await SendAndLogAsync(
+                    request,
+                    HttpCompletionOption.ResponseHeadersRead,
+                    token).ConfigureAwait(false));
+
+            switch (responseBox.Value.StatusCode)
+            {
+                case HttpStatusCode.OK:
+                    ValidateContentType(responseBox.Value, ContentTypes.ApplicationOctetStream);
+                    return await responseBox.GetAndReleaseAsync(ApiStreamHolder.CreateAsync).ConfigureAwait(false);
```

Comment:
Don't you already ConfigureAwait inside GetAndReleaseAsync?

Reply:
We should technically call ConfigureAwait(false) for any Task that is awaited... except for Tasks that are directly in the context of an xUnit test (so that the continuations are resumed on the xUnit SynchronizationContext). So everything should be calling ConfigureAwait(false) unless it is directly in the test method.